src/include/execution/ast/builtins.h

```diff
@@ -312,6 +312,20 @@ namespace noisepage::execution::ast {
   F(OffsetOf, offsetOf)                                                 \
   F(PtrCast, ptrCast)                                                   \
                                                                         \
+  /* Low-level Atomics*/                                                \
```

Comment:
Do you need a builtin for each type, or is it sufficient to have just one of each type and generate the appropriate bytecode based on the type of input after SA?

Reply:
Do you have a good example builtin I can mimic for that behavior?  I agree that would be nicer but didn't see a clear way to do that (ties into the TODO of making the compare-exchange ops not require the type to be `uint` as well).

Reply:
Take a look at the `@hash()` builtins.